### ISSUE TITLE
Add support for icon-only buttons

### ISSUE THREAD

**Issue by ricknout** [material-components/material-components-android#1283]
It would be useful if MDC-Android included functionality to implement icon-only buttons. This type of square style can be seen in the toggle button section of the guidelines - https://material.io/components/buttons#toggle-button - but it is fairly common to use these as standalone components too.

### Current means of implementation
Currently, there are a couple less-than-ideal ways to implement this:
- Using an `ImageButton`, `android:background="?attr/selectableItemBackgroundBorderless"` as well as a theme overlay (with `android:theme="..."`) in order to adjust `colorControlHighlight` which is the tint of the ripple
- Using a `MaterialButton` with a custom style similar to this:
```
<style name="Widget.App.Button.TextButton.IconOnly" parent="Widget.MaterialComponents.Button.TextButton">
    <item name="iconPadding">0dp</item>
    <item name="android:insetTop">0dp</item>
    <item name="android:insetBottom">0dp</item>
    <item name="android:paddingLeft">12dp</item>
    <item name="android:paddingRight">12dp</item>
    <item name="android:minWidth">48dp</item>
    <item name="android:minHeight">48dp</item>
</style>
```

### Desired solution
Ideally, MDC-Android could ship two means of achieving this:
1) A dedicated class - `IconButton` - that extends from `ImageButton` but adds support for convenient attrs like `shapeAppearance`, `backgroundTint`, `rippleColor`, etc. This would be useful in standalone cases and would not pull in all of the `TextView`-related APIs. 
2) Icon-only `MaterialButton` styles for each of the 4 variants. This would be useful when using `MaterialButtonToggleGroup`.
- `Widget.MaterialComponents.Button.IconOnly`
- `Widget.MaterialComponents.Button.UnelevatedButton.IconOnly`
- `Widget.MaterialComponents.Button.OutlinedButton.IconOnly`
- `Widget.MaterialComponents.Button.TextButton.IconOnly`

### Implementations in other libraries/frameworks:
- Jetpack Compose: [IconButton](https://android.googlesource.com/platform/frameworks/support/+/androidx-master-dev/ui/ui-material/src/main/java/androidx/ui/material/IconButton.kt)
- Flutter: [IconButton](https://api.flutter.dev/flutter/material/IconButton-class.html)
 
### Related issues
https://github.com/material-components/material-components-android/issues/272

**Comment by LouisCAD** [comment]
Being able to do so programmatically would be great. We had enough of xml lock-in. This makes adapting the look based on runtime factors such as window/device size, orientation, server-driven configuration, etc inconvenient, or impossible.

Related to the fact that we cannot set insets programmatically: https://github.com/material-components/material-components-android/issues/526

**Comment by drchen** [comment]
The icon-only styles have been released with 1.7 versions. : )
